Clarify futher  the order for a rolling upgrade

DaveCTurner opened the pull request:
Expands the "master-ineligible then master-eligible" sentence into a list and
specifies that within these subsets the order doesn't matter.

DaveCTurner committed:
Clarify futher  the order for a rolling upgrade

Expands the "master-ineligible then master-eligible" sentence into a list and
specifies that within these subsets the order doesn't matter.

elasticmachine commented:
Pinging @elastic/es-docs (>docs)

elasticmachine commented:
Pinging @elastic/es-distributed (:Distributed/Cluster Coordination)

eedugon reviewed (approved):
LGTM

jrodewig commented on docs/reference/upgrade/rolling_upgrade.asciidoc:
```diff
@@ -7,11 +7,23 @@ a time so upgrading does not interrupt service. Running multiple versions of
-upgrade the master-eligible nodes last then all the other nodes will not be
-running an older version and so they will be able to join the cluster.
+It is strongly recommended to upgrade the nodes in your cluster in the
+following order:
```
I'd use direct voice here. It makes the sentence a bit stronger.

> We strongly recommend you upgrade your cluster's nodes in the following order:

jrodewig commented on docs/reference/upgrade/rolling_upgrade.asciidoc:
```diff
@@ -7,11 +7,23 @@ a time so upgrading does not interrupt service. Running multiple versions of
+It is strongly recommended to upgrade the nodes in your cluster in the
+following order:
+
+. Nodes that are not master-eligible. You can retrieve a list of these nodes
```
I would link to master-eligible here.

```suggestion
. Nodes that are not <<master-node,master-eligible>>. You can retrieve a list of these nodes
```

jrodewig commented on docs/reference/upgrade/rolling_upgrade.asciidoc:
```diff
@@ -7,11 +7,23 @@ a time so upgrading does not interrupt service. Running multiple versions of
+following order:
+
+. Nodes that are not master-eligible. You can retrieve a list of these nodes
+with `GET /_nodes/master:false`, or by finding all the nodes which are
```
This comma is unneeded.
```suggestion
with `GET /_nodes/master:false` or by finding all the nodes
```

jrodewig commented on docs/reference/upgrade/rolling_upgrade.asciidoc:
```diff
@@ -7,11 +7,23 @@ a time so upgrading does not interrupt service. Running multiple versions of
+
+. Nodes that are not master-eligible. You can retrieve a list of these nodes
+with `GET /_nodes/master:false`, or by finding all the nodes which are
+configured with `node.master: false`. You may upgrade this subset of your nodes
```
I don't think "your" is needed here.

> configured with `node.master: false`. You may upgrade this subset of nodes

jrodewig commented on docs/reference/upgrade/rolling_upgrade.asciidoc:
```diff
@@ -7,11 +7,23 @@ a time so upgrading does not interrupt service. Running multiple versions of
+configured with `node.master: false`. You may upgrade this subset of your nodes
+in any order.
+
+. Nodes that are master-eligible, which is all the remaining nodes. You can
```
I'd reword this slightly:

> Master-eligible nodes, which are all remaining nodes in the cluster.

jrodewig commented on docs/reference/upgrade/rolling_upgrade.asciidoc:
```diff
@@ -7,11 +7,23 @@ a time so upgrading does not interrupt service. Running multiple versions of
+
+. Nodes that are master-eligible, which is all the remaining nodes. You can
+retrieve a list of these nodes with `GET /_nodes/master:true`. You may upgrade
+this subset of your nodes in any order.
```
"your" is unneeded.

> this subset of nodes in any order.

jrodewig commented on docs/reference/upgrade/rolling_upgrade.asciidoc:
```diff
@@ -7,11 +7,23 @@ a time so upgrading does not interrupt service. Running multiple versions of
+cluster with an older master, but older nodes cannot always join a cluster with
+a newer master. By upgrading the master-eligible nodes last you ensure that all
+the other nodes are running the newer version first and so they will be able to
+join the cluster whether the master-eligible nodes have been upgraded or not.
```
I believe this sentence is stronger without "The reason for ..." and some rewording:

> Newer nodes can always join a cluster with an older master, but older nodes cannot always join a cluster with
a newer master. This upgrade approach ensure all other nodes are first running a newer version. This means they can join the cluster, even if the cluster's master-eligible nodes have not been upgraded.

jrodewig reviewed (approved):
This looks pretty good. I left some suggestions for rewording which you can take or leave as you see fit.

I think all the crucial info is here. Thanks again for raising this.

inqueue reviewed (approved):
LGTM

crisdarocha commented:
LGTM, specially with the comments from @jrodewig .

crisdarocha reviewed (approved):
LGTM specially with @jrodewig changes

richiejarvis reviewed (dismissed):
Hi All,

I do not think that having the phrase "You may upgrade this subset of your nodes in any order" after each one is clear enough.  How about putting this more general text above?

"The order in which you upgrade each node in each subset is not important."

Commit:
Modified the text a little

Removed the duplicated text from the 2 "groups" of nodes

jasontedor commented on docs/reference/upgrade/rolling_upgrade.asciidoc:
```diff
@@ -7,11 +7,20 @@ a time so upgrading does not interrupt service. Running multiple versions of
-they may form a cluster that nodes of older versions cannot join. If you
-upgrade the master-eligible nodes last then all the other nodes will not be
-running an older version and so they will be able to join the cluster.
+It is strongly recommended to upgrade the nodes in your cluster by grouping the nodes into two groups, as shown below.  Nodes within each group may be upgraded in any order, as long as the last group is the master-eligable group:
```
Nit: please wrap at 80 columns.

jasontedor commented on docs/reference/upgrade/rolling_upgrade.asciidoc:
```diff
@@ -7,11 +7,20 @@ a time so upgrading does not interrupt service. Running multiple versions of
+retrieve a list of these nodes with `GET /_nodes/master:true`.
+
+By upgrading in this order it ensures that unforeseen circumstances do not break the upgrade.  
+Newer nodes can always join a cluster with an older master, but older nodes cannot always join a cluster with
```
Nit: please wrap at 80 columns.

jasontedor reviewed (commented):
I left a couple of nits about the column width.

DaveCTurner committed:
Merge branch 'master' into 2020-02-28-further-rolling-upgrade-docs

DaveCTurner committed:
Review suggestions

jasontedor commented on docs/reference/upgrade/rolling_upgrade.asciidoc:
```diff
@@ -7,11 +7,27 @@ a time so upgrading does not interrupt service. Running multiple versions of
-they may form a cluster that nodes of older versions cannot join. If you
-upgrade the master-eligible nodes last then all the other nodes will not be
-running an older version and so they will be able to join the cluster.
+We strongly recommend you upgrade you divide your cluster's nodes into the
```
"We strongly recommend you upgrade you divide your cluster's nodes" -> "We strongly recommend that when you upgrade you divide your cluster's nodes"

DaveCTurner replied:
Good catch, thanks

jasontedor reviewed (commented):
Sorry, one sentence needs a little extra love and care.

DaveCTurner committed:
Dedup

DaveCTurner commented:
Thanks all. @richiejarvis you requested changes but then pushed changes directly to the branch  rather than leaving them as suggestions. I've fixed the formatting and addressed the remaining  comments.

DaveCTurner committed:
Bah
